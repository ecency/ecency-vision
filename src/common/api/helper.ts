import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => `${defaults.base}${endpoint}`;

export const apiBaseImage = (endpoint: string): string => `${defaults.imageServer}${endpoint}`;
