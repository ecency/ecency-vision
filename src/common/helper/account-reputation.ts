const isHumanReadable = (input: number): boolean => {
  return Math.abs(input) > 0 && Math.abs(input) <= 100;
};

export default (input: string | number): number => {
  if (typeof input === 'number' && isHumanReadable(input)) {
    return Math.floor(input);
  }

  if (typeof input === 'string') {
    input = Number(input);

    if (isHumanReadable(input)) {
      return Math.floor(input);
    }
  }

  if (input === 0) {
    return 25;
  }

  let neg = false;

  if (input < 0) neg = true;

  let reputationLevel = Math.log10(Math.abs(input));
  reputationLevel = Math.max(reputationLevel - 9, 0);

  if (reputationLevel < 0) reputationLevel = 0;

  if (neg) reputationLevel *= -1;

  reputationLevel = reputationLevel * 9 + 25;

  return Math.floor(reputationLevel);
};
