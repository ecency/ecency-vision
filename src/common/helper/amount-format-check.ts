export default (v: string) => /^\d{1,3}(,?\d{3})*(\.(\d{3},?)*\d{1,3})?$/.test(v);
