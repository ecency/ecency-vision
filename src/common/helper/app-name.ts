import stripTags from "../util/strip-tags";

interface ObjInput {
  name?: string;
}

export default (input: string | null | undefined | ObjInput) => {
  if (!input) {
    return "";
  }

  if (typeof input === "string") {
    return stripTags(input);
  }

  if (typeof input === "object" && input.name !== undefined) {
    return stripTags(input.name);
  }

  return "";
};
