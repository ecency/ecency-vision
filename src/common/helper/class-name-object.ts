export function classNameObject(obj: Record<string, boolean | undefined>): string {
  return Object.entries(obj)
    .filter(([className, isActive]) => !!isActive)
    .map(([className]) => className)
    .join(" ");
}
