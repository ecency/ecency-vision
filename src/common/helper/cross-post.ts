import {postBodySummary} from '@ecency/render-helper';
import {Entry} from '../store/entries/types';

export const crossPostMessage = (body: string) => {
  const crossPostRegex =
    /^This is a cross post of \[@(.*?)\/(.*?)\]\(\/.*?@.*?\/.*?\) by @.*?\.<br>/;

  if (body.match(crossPostRegex)) {
    const message = body.replace(crossPostRegex, '');
    return postBodySummary(message);
  }

  return null;
};

export const makeCrossPostMessage = (
  entry: Entry,
  poster: string,
  message: string,
) => {
  return `This is a cross post of [@${entry.author}/${entry.permlink}](/${entry.category}/@${entry.author}/${entry.permlink}) by @${poster}.<br><br>${message}`;
};
