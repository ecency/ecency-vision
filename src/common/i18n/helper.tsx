import React from "react";

import xss from "xss";

import { _t } from "./index";

export const HTML_WHITELIST = {
  a: ["class", "href", "target", "rel"],
  span: ["class"],
  h3: [],
  p: [],
  strong: [],
  b: [],
  img: ["src", "alt"],
  ul: [],
  li: [],
  br: []
};

export const safeHtml = (html: string): string => {
  return xss(html, {
    whiteList: HTML_WHITELIST,
    stripIgnoreTagBody: true,
    stripIgnoreTag: true,
    css: false
  });
};

interface Props {
  k: string;
  args?: {};
  children: JSX.Element;
}

export class Tsx extends React.Component<Props> {
  render() {
    const { k, args } = this.props;

    const { children } = this.props;

    const html = safeHtml(_t(k, args));

    return React.cloneElement(children, { dangerouslySetInnerHTML: { __html: html } });
  }
}
