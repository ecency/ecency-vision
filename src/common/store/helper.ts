import {Store} from 'redux';

import i18n from 'i18next';

import {AppState} from './index';
import {ActiveUser, UserPoints} from './active-user/types';
import {
  loginAct as loginActiveUser,
  logoutAct as logoutActiveUser,
  updateAct as updateActiveUserAct,
} from './active-user';

import {getAccount, getDynamicProps} from '../api/hive';
import {getPoints, usrActivity, getPromotedEntries} from '../api/private-api';
import {reloadAct as reloadUsers} from './users';
import {fetchedAct as loadDynamicProps} from './dynamic-props';
import {fetchedAct as entriesFetchedAct} from './entries';
import {
  setCurrencyAct as setCurrency,
  muteNotificationsAct as muteNotifications,
  setLangAct as setLang,
  setNsfwAct as setNsfw,
} from './global';

import {getCurrencyRate} from '../api/misc';

import currencies from '../constants/currencies.json';

import * as ls from '../../common/util/local-storage';

import currencySymbol from '../../common/helper/currency-symbol';

import {AppWindow} from '../../client/window';

declare var window: AppWindow;

export const activeUserMaker = (
  name: string,
  points: string = '0.000',
  uPoints: string = '0.000',
): ActiveUser => {
  return {
    username: name,
    data: {name},
    points: {
      points,
      uPoints,
    },
  };
};

export const activeUserUpdater = async (store: Store<AppState>) => {
  const state = store.getState();
  if (state.activeUser) {
    const {username} = state.activeUser;

    let account;
    try {
      account = await getAccount(username);
    } catch (e) {
      return;
    }

    let points: UserPoints;
    try {
      const p = await getPoints(username);
      points = {points: p.points, uPoints: p.unclaimed_points};
    } catch (e) {
      points = {
        points: '0.000',
        uPoints: '0.000',
      };
    }

    store.dispatch(updateActiveUserAct(account, points));
  }
};

export const syncActiveUser = (store: Store<AppState>) => {
  const state = store.getState();

  const activeUser = ls.get('active_user');

  // logout
  if (!activeUser && state.activeUser) {
    store.dispatch(logoutActiveUser());
    return;
  }

  // login
  if (activeUser && !state.activeUser) {
    store.dispatch(loginActiveUser());
    activeUserUpdater(store).then();
    return;
  }

  // switch
  if (
    activeUser &&
    state.activeUser &&
    activeUser !== state.activeUser.username
  ) {
    store.dispatch(loginActiveUser());
    activeUserUpdater(store).then();
    return;
  }
};

export const clientStoreTasks = (store: Store<AppState>) => {
  // To use in places where we can't access to store
  window.usePrivate = store.getState().global.usePrivate;

  // Initial state from browser's local storage
  store.dispatch(reloadUsers());
  store.dispatch(loginActiveUser());

  // Load dynamic props
  getDynamicProps().then(resp => {
    store.dispatch(loadDynamicProps(resp));
  });

  // Update active user in interval
  activeUserUpdater(store).then();
  setInterval(() => {
    activeUserUpdater(store).then();
  }, 60 * 1000);

  // Active user sync between tabs / windows
  setInterval(() => {
    syncActiveUser(store);
  }, 2000);

  // Do check-in in interval
  const checkIn = () => {
    const state = store.getState();
    if (state.activeUser) {
      usrActivity(state.activeUser?.username!, 10).then();
    }
  };

  // wait for initial user sync
  setTimeout(() => {
    checkIn();
  }, 5000);
  setInterval(checkIn, 1000 * 60 * 15 + 8);

  // Inject / update promoted entries to store
  const promotedEntries = () => {
    getPromotedEntries().then(r => {
      store.dispatch(entriesFetchedAct('__promoted__', r, false));
    });
  };
  promotedEntries();
  setInterval(promotedEntries, 1000 * 60 * 5);

  // Currency
  const currency = ls.get('currency');
  if (currency && currencies.find(x => x.id === currency)) {
    const symbol = currencySymbol(currency);
    getCurrencyRate(currency).then(rate => {
      store.dispatch(setCurrency(currency, rate, symbol));
    });
  }

  // Notifications
  if (ls.get('notifications') === false) {
    store.dispatch(muteNotifications());
  }

  // Language
  const lang = ls.get('lang');
  if (lang) {
    if (store.getState().global.lang !== lang) {
      i18n.changeLanguage(lang).then(() => {
        store.dispatch(setLang(lang));
      });
    }
  }

  // NSFW
  if (ls.get('nsfw') === true) {
    store.dispatch(setNsfw(true));
  }
};
