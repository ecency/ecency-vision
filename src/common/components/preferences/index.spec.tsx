import React from "react";

import { Preferences } from "./index";

import renderer from "react-test-renderer";

import { activeUserInstance, globalInstance } from "../../helper/test-helper";

it("(1) Default render", () => {
  const props = {
    global: globalInstance,
    muteNotifications: () => {},
    unMuteNotifications: () => {},
    setCurrency: () => {},
    setLang: () => {},
    setNsfw: () => {}
  };

  const component = renderer.create(<Preferences {...props} activeUser={activeUserInstance} />);
  expect(component.toJSON()).toMatchSnapshot();
});
