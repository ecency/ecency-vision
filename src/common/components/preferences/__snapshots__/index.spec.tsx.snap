// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  className="preferences"
>
  <div
    className="preferences-header"
  >
    Preferences
  </div>
  <div
    className="form-row"
  >
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Notification alerts
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          type="text"
          value={1}
        >
          <option
            value={1}
          >
            On
          </option>
          <option
            value={0}
          >
            Off
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Currency
        </label>
        <select
          className="form-control"
          disabled={false}
          onChange={[Function]}
          type="text"
          value="usd"
        >
          <option
            value="btc"
          >
            BTC
          </option>
          <option
            value="hbd"
          >
            Hive Based Dollar
          </option>
          <option
            value="usd"
          >
            USD
          </option>
          <option
            value="eur"
          >
            EUR
          </option>
          <option
            value="brl"
          >
            BRL
          </option>
          <option
            value="rub"
          >
            RUB
          </option>
          <option
            value="gbp"
          >
            GBP
          </option>
          <option
            value="jpy"
          >
            JPY
          </option>
          <option
            value="krw"
          >
            KRW
          </option>
          <option
            value="inr"
          >
            INR
          </option>
          <option
            value="cny"
          >
            CNY
          </option>
          <option
            value="uah"
          >
            UAH
          </option>
          <option
            value="sek"
          >
            SEK
          </option>
          <option
            value="try"
          >
            TRY
          </option>
          <option
            value="cad"
          >
            CAD
          </option>
          <option
            value="chf"
          >
            CHF
          </option>
          <option
            value="aud"
          >
            AUD
          </option>
          <option
            value="nok"
          >
            NOK
          </option>
          <option
            value="pln"
          >
            PLN
          </option>
          <option
            value="php"
          >
            PHP
          </option>
          <option
            value="idr"
          >
            IDR
          </option>
          <option
            value="zar"
          >
            ZAR
          </option>
          <option
            value="thb"
          >
            THB
          </option>
          <option
            value="pkr"
          >
            PKR
          </option>
          <option
            value="vnd"
          >
            VND
          </option>
          <option
            value="ngn"
          >
            NGN
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Language
        </label>
        <select
          className="form-control"
          disabled={false}
          onChange={[Function]}
          type="text"
          value="en-US"
        >
          <option
            value="en-US"
          >
            English
          </option>
          <option
            value="es-ES"
          >
            Español
          </option>
          <option
            value="hi-IN"
          >
            हिन्दी
          </option>
          <option
            value="it-IT"
          >
            Italiano
          </option>
          <option
            value="id-ID"
          >
            Bahasa Indonesia
          </option>
          <option
            value="pt-PT"
          >
            Português
          </option>
          <option
            value="sr-CS"
          >
            Srpski
          </option>
          <option
            value="fi-FI"
          >
            Suomi
          </option>
          <option
            value="uk-UA"
          >
            Українська
          </option>
          <option
            value="bg-BG"
          >
            Български
          </option>
          <option
            value="ru-RU"
          >
            Русский
          </option>
          <option
            value="uz-UZ"
          >
            O'zbekcha
          </option>
          <option
            value="zh-CN"
          >
            简体字
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="form-row"
  >
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Show NSFW content
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          type="text"
          value={0}
        >
          <option
            value={1}
          >
            On
          </option>
          <option
            value={0}
          >
            Off
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Referral link
        </label>
        <div
          className="mb-3 input-group"
          onClick={[Function]}
        >
          <input
            className="text-primary pointer form-control"
            disabled={true}
            value="https://ecency.com/signup?referral=foo"
          />
          <div
            className="input-group-append"
          >
            <button
              className="copy-to-clipboard btn btn-primary btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                fill="#FFFFFF"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Theme Settings
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          type="text"
        >
          <option
            value="system"
          >
            System Default
          </option>
          <option
            value="day"
          >
            Day mode
          </option>
          <option
            value="night"
          >
            Night mode
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-12 col-lg-6"
    >
      Entry Item Options
       
    </div>
    <hr />
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Show Self Vote flag
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          type="text"
          value="true"
        >
          <option
            value="true"
          >
            On
          </option>
          <option
            value="false"
          >
            Off
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Show Hive Power/Hive Dollar Breakdown
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          type="text"
          value="true"
        >
          <option
            value="true"
          >
            On
          </option>
          <option
            value="false"
          >
            Off
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Low maximum accepted reward threshold
        </label>
        <input
          className="form-control"
          defaultValue={100}
          onChange={[Function]}
          type="text"
        />
      </div>
    </div>
    <div
      className="col-xl-4 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Show front end name
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          type="text"
          value="true"
        >
          <option
            value="true"
          >
            On
          </option>
          <option
            value="false"
          >
            Off
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-xl-12 col-lg-6"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Footer for your posts
        </label>
        <textarea
          className="the-editor accepts-emoji form-control form-control"
          defaultValue="Posted with [proof of brain](https://proofofbrain.blog)"
          id="the-footer-editor"
          onChange={[Function]}
          placeholder="Tell your story..."
          spellCheck={true}
        />
      </div>
    </div>
    <div
      className="col-xl-12"
    >
      <label
        className="form-label"
      >
        Footer Preview
      </label>
      <br />
      <div
        className="preview-body markdown-view"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Posted with <a href=\\"https://proofofbrain.blog\\" class=\\"markdown-external-link\\" target=\\"_blank\\" rel=\\"noopener\\">proof of brain</a></p>",
          }
        }
      />
    </div>
  </div>
</div>
`;
